### PR TITLE
refactor: split react.test.tsx in multiple files

### LINKED ISSUE #1661: Refactor `react.test.tsx`
Related to the [`next`](https://github.com/trpc/trpc/tree/next) branch.

- It's a big mess. 
- It should be split into logical files.

### PR DESCRIPTION
**Pull request by joaosamouco**
This MR just splits `react.test.tsx` into multiple logical files.
Maybe more improvements could be made but those changes would be hard to track in this PR. I believe this is a good first step and we can work from there.

Closes #1661

### CODE REVIEW

File: packages/server/test/interop/react/__testHelpers.tsx [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+});
+
+type Context = {};
+export type Post = {
+  id: string;
+  title: string;
+  createdAt: number;
+};
+
+export function createAppRouter() {
```
Comment (KATT) [inline packages/server/test/interop/react/__testHelpers.tsx:39]: This is the v9 way of doing stuff.

```suggestion
/** @deprecated **/
export function createLegacyAppRouter() {
```

**Review by KATT — COMMENTED** [review]
Great!

We can use this to have a foundation for v10 test suite. As you probably noticed, this is all in the `/interop`-folder which is legacy (current v9) stuff

File: packages/server/test/interop/react/__testHelpers.tsx (hunk 1 above)
Comment (joaosamouco) [reply packages/server/test/interop/react/__testHelpers.tsx]: Added your suggestion and renamed everything accordingly :+1:

**Comment by KATT** [comment]
@all-contributors add @joaosamouco for tests

**Pull request merged**